### ISSUE TITLE
ambiguity regarding vkCmdBindVertexBuffers offsets

### ISSUE THREAD

**Issue by timon37** [KhronosGroup/Vulkan-Docs#1277]
Hi, recently we ran into an issue with broken drawcalls and/or gpu hang on amd hardware (os+gpu dependent, details below).
Tracked it down to vkCmdBindVertexBuffers offset requiring alignment (in testing it seems 4 was required).

Reading through the spec I couldn't really find much about this (there's an explanation regarding uniform/storage/texel buffers, or vkCmdBindIndexBuffer offset).
So I'm not sure whether it should be the alignment of the largest attribute of a vertex, or all attributes should have natural alignment, or if it should be taken from BufferMemoryRequirements, or something else.

In practice from my experiments: BufferMemoryRequirements seems definitely overkill, e.g. nvidia+linux reports 256 but it was fine with alignment of 2 (maybe even 1 not sure if I tested that explicitly).
On amd+linux it's 16 (minor overkill), and amd+windows reported 4 (which seems to match my test, but I didn't test 8byte vertex attributes).

On windows I only got broken drawcalls with either a 580 or 5700.
On linux (radv) 580 seemed fine with lower alignment, however 5700 would hang and crash Xorg.

I didn't have a quick way to test 8byte values in the vertex buffer (sorry).

**Comment by jeffbolznv** [comment]
This is mostly answered in this section:

```
[[fxvertex-input-extraction]]
For each attribute, raw data is extracted starting at `attribAddress` and is
converted from the sname:VkVertexInputAttributeDescription's pname:format to
either floating-point, unsigned integer, or signed integer based on the base
type of the format; the base type of the format must: match the base type of
the input variable in the shader.
If pname:format is a packed format, `attribAddress` must: be a multiple of
the size in bytes of the whole attribute data type as described in
<<formats-packed,Packed Formats>>.
Otherwise, `attribAddress` must: be a multiple of the size in bytes of the
component type indicated by pname:format (see <<formats,Formats>>).
If the format does not include G, B, or A components, then those are filled
with [eq]#(0,0,1)# as needed (using either 1.0f or integer 1 based on the
format) for attributes that are not 64-bit data types.
The number of components in the vertex shader input variable need not
exactly match the number of components in the format.
If the vertex shader has fewer components, the extra components are
discarded.
```

We're going to investigate whether this can be turned into VUs somewhere.

**Comment by oddhack** [comment]
This should be fixed in the 1.4.304 spec update.
